Account Page

jordles opened the pull request:
PR Update! Account Page

* Left Back Arrow - returns to the previous page, based on history
* Trash Button - removes user information and destroys session and cookies with the `auth/logout` route
* Image add - We currently do not have a concrete plan on how we store images. Only image selection is the only feature available. 
* Username/Email - both editable, focused, and highlighted when interacting with pencil icon on clicking on the input field 
* Password - 2 fields and validation rules included
* Update Button - Updates the user information to database 

* auth/logout - updated to destory session and remove cookies. Removes redirect since React expects a response, not a browser-level navigation. I've moved this to the frontend code instead. 

Miscellaneous 

I added a npm script to connect to my psql database to offset manually typing the command: `docker exec -it postgres_container psql -U <POSTGRES_USER> -d <POSTGRES_DB>`

I'm using the dotenv-cli package to run my environment variable called `CONNECT` on the npm script. `"connect": "dotenv -e .env -- sh -c \"$CONNECT\"",`


---

Since the image functionality is not fully done, i wont be linking this PR to a project board task just yet. To be discussed further in the future.

robinallenaz commented:
Looks good and everything seems to be working, but email validation is missing.

jordles commented:
> Looks good and everything seems to be working, but email validation is missing. Also, when a user is already signed in and tries to sign in again, the error "failed to deserialize user session" appears.


UPDATE: 

- Implemented a modular function for validation to be reused for email and password (accountProfile only, if Login component wants to be using these components please let me know Ken on how to proceed) 
- Made a new error message component
- Fixed the deserializing issues regarding users. <ins>Important to note: since were in a testing environment, deleting volumes on docker will remove the user information from our database causing a deserializing issue. Just log back in or remove the cookie<ins>
- Going to the login url will redirect users back to the home page if theyre already logged in
- Implemented Toastify messages for update and delete user accounts. (will plan for improvements to make it more modular)

robinallenaz commented:
Everything is looking good on my end! I received errors at first but after running npm i in the frontend directory, everything works as described.

jordles commented:
## UPDATE

I made my customcomponent modular `CustomComponent.jsx`

If you want to use my customcomponent (layout html will be the same) for other uses, use the available properties in the data object when you call toast. The prop names include `title, content, function, icon` Icons use google's Material Symbols

```
data: {
  title: "Confirm Account Deletion",
  content: (
    <>
      This will permanently delete your account and all associated data.<br />
      Are you sure?
    </>
  ),
  function: handleDelete,
  icon: "delete"
}
```

The `ToastComponent` component is on the root level `App.jsx`. It will read any/all toast event calls and render the toastify messages onscreen.

ntguru5 reviewed (approved):
tested updating and removing account
